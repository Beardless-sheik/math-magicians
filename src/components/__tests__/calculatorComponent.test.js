import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import CalculatorUI from '../calculatorComponent/Calculator';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CalculatorUI />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the product of two number when interacting with the UI', () => {
    render(<CalculatorUI />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByTestId('result');
    expect(result.innerHTML).toBe('21');
  });
});
