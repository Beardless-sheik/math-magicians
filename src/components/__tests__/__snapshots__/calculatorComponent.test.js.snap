// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="grid1 calculatorValueContainer"
  >
    <div
      className="calculatorValue"
      data-testid="result"
    >
      0
      
      
    </div>
  </div>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     AC  
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     +/- 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     % 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     + 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     7 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
      8 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     9 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     x 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     4 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     5 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     6 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     - 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     1 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     2 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     3 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     + 
  </button>
  <button
    className="gridLightGreyContainer zeroValueContainer"
    onClick={[Function]}
    type="button"
  >
     0 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     . 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     = 
  </button>
</div>
`;

exports[`Calculator page component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="grid1 calculatorValueContainer"
  >
    <div
      className="calculatorValue"
      data-testid="result"
    >
      0
      
      
    </div>
  </div>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     AC  
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     +/- 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     % 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     + 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     7 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
      8 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     9 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     x 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     4 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     5 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     6 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     - 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     1 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     2 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     3 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     + 
  </button>
  <button
    className="gridLightGreyContainer zeroValueContainer"
    onClick={[Function]}
    type="button"
  >
     0 
  </button>
  <button
    className="gridLightGreyContainer"
    onClick={[Function]}
    type="button"
  >
     . 
  </button>
  <button
    className="gridOrangeContainer"
    onClick={[Function]}
    type="button"
  >
     = 
  </button>
</div>
`;
